package.json

```diff
@@ -26,14 +26,16 @@
 		"php:lint": "composer phpcs:lint",
 		"prepare": "husky",
 		"reformat-files": "prettier --ignore-path .eslintignore --write \"**/*.{js,jsx,cjs,mjs,ts,tsx,svelte,json,json5}\"",
+		"typecheck": "pnpm --recursive --if-present --no-bail run typecheck",
 		"version-packages": "bash ./tools/version-packages.sh"
 	},
 	"devDependencies": {
 		"eslint": "9.22.0",
 		"husky": "9.1.7",
 		"jetpack-cli": "workspace:*",
 		"jetpack-js-tools": "workspace:*",
-		"stylelint": "16.16.0"
+		"stylelint": "16.16.0",
+		"typescript": "5.8.2"
```

Comment:
Let's not add this here please. It means every package can continue to forget to add the needed dependency.

Reply:
Then we should enforce that every package that has a `tsconfig.json` must have `typescript` as a dependency.

Reply:
Reverted that change. Thanks.

Reply:
Is typescript as a dependency really needed for a tsconfig? Or would it be better to enforce that a package with a `tsconfig.json` should have a `typecheck` script defined (which would probably lead to a typescript dep being added to implement it)?

If you want to add such a check, `.github/files/lint-project-structure.sh` would be the place to add it.

Reply:
> Or would it be better to enforce that a package with a tsconfig.json should have a typecheck script defined

That will be better I think.

Reply:
That change probably needs more planning as it makes this change too big to be a quick fix. The reason is that when we implement that structure constraint, we first need to fix all those errors which arise after installing `typescript` for all of them.

May be we can delegate it to the teams that manage the packages. If there is no particular owner, we can do it in separate follow-up PRs.